mobile-app/app/components/BottomSheetTokenList.tsx

```diff
@@ -89,7 +93,8 @@ export const BottomSheetTokenList = ({
                     onButtonPress: navigateToScreen.onButtonPress,
                     collateralFactor: new BigNumber(item.factor ?? 0).times(100),
                     isAdd: true,
-                    vault
+                    vault,
+                    ...(isCollateralItem(item) && { collateralItem: item })
```

Comment:
Why are you spreading the return of this?

Reply:
Suggested to use this to @chloezxyy to be able to pass the collateralItem to ensure it won't be `undefined`, but it needs to check if the type is `CollateralItem` via ts typeguard so it won't affect other components using the BottomSheet that doesn't have CollateralItem e.g composite swap